command/state_pull.go

```diff
@@ -70,9 +70,13 @@ func (c *StatePullCommand) Help() string {
 	helpText := `
 Usage: terraform state pull [options]
 
-  Pull the state from its location and output it to stdout.
+  Pull the state from its location, upgrade the local copy, and output it
+  to stdout.
 
   This command "pulls" the current state and outputs it to stdout.
+  As part of this process, Terraform will upgrade the state format of the
+  local copy to the current version.
```

Comment:
@alisdair Sorry to be a little bit late chiming in here, but it might be clearer to be explicit about what we mean by "current version". I assume that's the version that's compatible with the locally-installed version of Terraform?

Reply:
That's correct. Terraform's state file reading process converts the state file into the latest version supported by the Terraform binary performing the operation.

I'm not sure how to more clearly express this than what's in the docs now, but I'd be happy to review/merge a PR if you have ideas!